ocp_resources/resource.py

```diff
@@ -1017,6 +1030,26 @@ def to_yaml(self):
         self.logger.info(f"\n{resource_yaml}")
         return resource_yaml
 
+    def hash_resource_dict(self, resource_dict: dict):
+        if hasattr(self, "keys_to_hash"):
+            return self.hash_resource_dict_rec(nested_dict=copy.deepcopy(resource_dict))
+        return resource_dict
+
+    def hash_resource_dict_rec(self, nested_dict: dict):
+        if not isinstance(nested_dict, dict):
+            return nested_dict
+
+        for key in nested_dict.keys():
+            if key in self.keys_to_hash:
+                if isinstance(nested_dict[key], dict):
+                    nested_dict[key] = {_key: "***" for _key in nested_dict[key].keys()}
+                else:
+                    nested_dict[key] = "***"
+            else:
+                nested_dict[key] = self.hash_resource_dict_rec(nested_dict[key])
+
+        return nested_dict
```

Comment:
should be saved in `self.hashed_resource_dict` and not called every time this is needed

Reply:
currently for each CRUD method logs we pass a different data dict: Resource `delete` logs `self.instance.to_dict()`, `update` logs a given `resource_dict` and `create logs `self.res`.
Should we use a single attribute of resource as `self.hashed_resource_dict` (e.g. hash `self.instance.to_dict()) ?